File: codecamp/src/main/java/com/gdogaru/codecamp/model/Codecamp.java
/*
 * Copyright (C) 2008 Gabriel Dogaru (gdogaru@gmail.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.gdogaru.codecamp.model;

import java.io.Serializable;
import java.util.List;

/**
 * Created by Gabriel Dogaru (gdogaru@gmail.com)
 */
public class Codecamp extends EventSummary implements Serializable {

    private List<SponsorshipPackage> sponsorshipPackages;
    private List<Sponsor> sponsors;
    private List<Schedule> schedules;
    private List<Speaker> speakers;

    public List<Speaker> getSpeakers() {
        return speakers;
    }

    public void setSpeakers(List<Speaker> speakers) {
        this.speakers = speakers;
    }

    public List<SponsorshipPackage> getSponsorshipPackages() {
        return sponsorshipPackages;
    }

    public void setSponsorshipPackages(List<SponsorshipPackage> sponsorshipPackages) {
        this.sponsorshipPackages = sponsorshipPackages;
    }

    public List<Sponsor> getSponsors() {
        return sponsors;
    }

    public void setSponsors(List<Sponsor> sponsors) {
        this.sponsors = sponsors;
    }

    public List<Schedule> getSchedules() {
        return schedules;
    }

    public void setSchedules(List<Schedule> schedules) {
        this.schedules = schedules;
    }

}

File: codecamp/src/main/java/com/gdogaru/codecamp/api/model/Codecamp.kt
/*
 * Copyright (c) 2019 Gabriel Dogaru - gdogaru@gmail.com
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *  *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

package com.gdogaru.codecamp.api.model

import org.threeten.bp.LocalDateTime
import java.io.Serializable
import java.util.*

/**
 * @author Gabriel Dogaru (gdogaru@gmail.com)
 */
class Codecamp : EventSummary(), Serializable {

    var sponsorshipPackages: List<SponsorshipPackage>? = null
    var sponsors: List<Sponsor>? = null
    var schedules: List<Schedule>? = null
    var speakers: List<Speaker>? = null

}

class EventList : ArrayList<EventSummary>()

open class EventSummary {
    var refId: Long = 0
    var title: String? = null
    var startDate: LocalDateTime? = null
    var endDate: LocalDateTime? = null
    var venue: Venue? = null
}
